ksonnet-gen/astext/astext.go

```diff
@@ -34,7 +34,7 @@ type Object struct {
 
 var (
 	// reFieldStr matches a field id that should be enclosed in quotes.
-	reFieldStr = regexp.MustCompile(`^[_A-Za-z]+[_A-Za-z0-9\-]*$`)
+	reFieldStr = regexp.MustCompile(`^([_A-Za-z0-9\.]?[A-Za-z0-9\-_\.]+(\.[A-Za-z0-9\-_]+)*)?$`)
```

Comment:
also matches leading numbers / all numbers in addition to stated purpose of the PR

Reply:
Yes, this is fine. Updated the PR with a test for this.